### ISSUE TITLE
The console dropdown menu supports regular expressions？

### ISSUE THREAD

**Issue by yinmin2020** [kubesphere/kubesphere#4117]
**What's it about?**
When you want to select the drop-down menu box in the console, you can search by regular expression or sort by the letters A-Z
**What's the reason why we need it?**
Can improve the development efficiency, shorten the search time
I believe this is an important feature for Kubesphere. There're a few use cases:
For example, if I have 200 services on a project, I have to manually screen each service when creating an application route, which is a waste of time
![image](https://user-images.githubusercontent.com/65068467/128137843-e02df953-631f-49cd-9998-0fb3baec27be.png)

**Comment by zryfish** [comment]
Agreed. Maybe we could make the dropdown menu editable, so suggestions can be given when typing, just like adding annotations to Routes.

<img width="936" alt="截屏2021-08-05 15 46 38" src="https://user-images.githubusercontent.com/3326354/128312030-72c88058-738e-4aae-a44a-e3a15d97cf5c.png">

**Comment by harrisonliu5** [comment]
change the Select component to the AutoComplete component. [the example code](https://github.com/kubesphere/console/blob/master/src/components/Inputs/RulePath/index.jsx#L88)

**Comment by liangyuanpeng** [comment]
Seems like this issue should be in the [kubesphere/console](https://github.com/kubesphere/console).
